tools/idr_download/idr_download_by_ids.xml

```diff
@@ -1,84 +1,151 @@
+            help="Set to zero to extend the rectangle maximally along the x-axis of the image" />
+            <param name="height" type="integer" value="0" min="0"
+            label="Region height"
+            help="Set to zero to extend the rectangle maximally along the y-axis of the image" />
+        </xml>
     </macros>
     <requirements>
-        <requirement type="package" version="@VERSION@">python-omero</requirement>
+        <requirement type="package" version="5.6.0">python-omero</requirement>
         <requirement type="package" version="3.1.3">matplotlib</requirement>
     </requirements>
+    <stdio>
+        <exit_code range="1:" />
+    </stdio>
     <command><![CDATA[
-        #if str($id_source.source) == 'link':
-            python -c 'print("${id_source.image_id_link}".split("?show=")[-1].replace("|", "\n"))'
+        mkdir downloads && cd downloads &&
+        #if str($image_ids.source) == 'link':
+            python -c 'print("${image_ids.id_spec}".replace(",", "|").split("?show=")[-1].replace("|", "\n"))'
             ## https://idr.openmicroscopy.org/webclient/?show=image-3426274|image-3426275|image-3426276|image-3426277
         #else:
-            cat ${id_source.image_id_data}
+            cat ${image_ids.id_spec}
         #end if
         | python '$__tool_directory__/idr_download_by_ids.py' 
-            -c $channel 
-            -f $frame
-            -z $z_section
-            --center $x_coord $y_coord $width $height
-            &> $out_log
+        #set $channel = str($channel).strip()
+        #if $channel:
+            -c $channel
+        #end if
+        -f $frame
+        -z $z_section
+        #if str($clip_image.select):
+            ${clip_image.select} ${clip_image.x_coord} ${clip_image.y_coord} ${clip_image.width} ${clip_image.height}
+        #end if
 
+        2> >(tee -a $out_log >&2)
     ]]></command>
 <inputs>
-        <conditional name="id_source">
-            <param name="source" type="select" label="How do you want to provide the image ids?">
-                <option value="link">As a IDR link</option>
-                <option value="data">Read image IDs from dataset</option>
+        <conditional name="image_ids">
+            <param name="source" type="select" label="How would you like to specify the ids of images to download?">
+                <option value="link">As text (comma-separated list of ids or a valid IDR link)</option>
+                <option value="dataset">As a dataset (one image ID per line)</option>
             </param>
             <when value="link">
-                <param name="image_id_link" type="text"
-                label="Paste your IDR hyperlink">
+                <param name="id_spec" type="text"
+                label="Image ids to download"
+                help="You can enter a single image-id, or a comma (or '|')-separated list of ids. Alternatively, you can paste here a link to an image selection obtained through the IDR webclient.">
```

Comment:
@wm75, just doing some testing. The option to paste a link is not (yet) available, right?

Reply:
It *is* available. You just have to visit https://idr.openmicroscopy.org/, select some images, then manually copy/paste the link available in the top-right corner.

Reply:
Truncated links should also work as long as they contain the image-id part.

Reply:
Right, because of the `.split("?show=")` thing. Got it, thanks.